src/s3/util/fileViewerManager.ts

```diff
@@ -78,86 +92,135 @@ export class S3FileViewerManager {
+            const edit = 'Open in edit mode'
+            const read = 'Try in read-only'
+            if (await showConfirmationMessage({ prompt, confirm: edit, cancel: read }, this.window)) {
+                return await this.openInEditMode(fileNode)
+            }
+        }
 
-        const newTab = this.activeTabs.get(fileLocation.fsPath) ?? new S3Tab(fileLocation, fileNode)
-        await newTab.openFileInReadOnly()
+        let tab: S3Tab | undefined
+        if (fileNode.file.sizeBytes! < SIZE_LIMIT) {
+            const pathToPreview = await this.arnToFsPath(this.toPreview!)
+            if (s3Uri.fsPath !== pathToPreview) {
+                return
+            }
+            tab =
+                this.activeTabs.get(pathToPreview) ??
+                ({ fileUri: fileLocation, s3Uri, editor: undefined, s3FileNode: fileNode, type, charset } as S3Tab)
+            await this.openFileGivenMode(tab, tab.s3Uri, true)
+            this.toPreview = undefined
+        } else {
+            tab =
+                this.activeTabs.get(fileLocation.fsPath) ??
+                ({ fileUri: fileLocation, s3Uri, editor: undefined, s3FileNode: fileNode, type, charset } as S3Tab)
+            await this.openFileGivenMode(tab, tab.s3Uri, false)
+        }
 
-        this.activeTabs.set(fileLocation.fsPath, newTab)
+        this.activeTabs.set(fileLocation.fsPath, tab)
     }
 
+    /**
+     * Given an S3FileNode or an URI, this function:
+     * Checks and creates a cache to store downloads
+     * Retrieves previously cached files on cache and
+     * Downloads file from S3 ands stores in cache
+     * Opens the tab on read-only with the use of an S3Tab, or shifts focus to an edit tab if any.
+     *
+     * @param uriOrNode to be opened
+     */
     public async openInEditMode(uriOrNode: vscode.Uri | S3FileNode): Promise<void> {
         if (this.promptOnEdit) {
             const message = localize(
                 'AWS.s3.fileViewer.warning.editStateWarning',
-                'Opening S3 file for editing. Saved changes will be written directly to the original location in S3.'
+                'You are now editing an S3 file. Saved changes will be uploaded to your S3 bucket.'
             )
 
+            const ok = localize('AWS.s3.fileViewer.ok', 'Ok')
             const dontShow = localize('AWS.s3.fileViewer.button.dismiss', "Don't show this again")
-            const help = localize('AWS.s3.fileViewer.button.help', 'Help')
+            const help = localize('AWS.generic.message.learnMore', 'Learn more')
 
-            this.window.showWarningMessage(message, dontShow, help).then(selection => {
+            this.window.showWarningMessage(message, ok, dontShow, help).then(selection => {
                 if (selection === dontShow) {
                     this.promptOnEdit = false
                 }
 
                 if (selection === help) {
-                    //add help section
+                    //TODO: add help section
                 }
```

Comment:
Hide the help button for now. No reason to include a nonfunctional button in the UI.